services/QuillLMS/spec/queries/snapshots/most_active_schools_query_spec.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Snapshots
+  describe MostActiveSchoolsQuery do
+    include_context 'Snapshots TopX CTE'
+
+    context 'big_query_snapshot', :big_query_snapshot do
+      let(:num_classrooms) { 11 }
+      let(:max_activity_session_count) { 20 }
+
+      let(:classroom_units) { classrooms.map { |classroom| create(:classroom_unit, classroom: classroom) } }
+
+      # For each classroom (each of which has a single classroom_unit), create activity_sessions for it, but creating one less for each subsequent classroom so that they'll have different relevant counts
+      let(:activity_sessions) do
+        classroom_units.map.with_index do |classroom_unit, i|
+          create_list(:activity_session, (max_activity_session_count - i), classroom_unit: classroom_unit)
+        end
+      end
+
+      let(:runner_context) {
+        [
+          classrooms,
+          teachers,
+          classrooms_teachers,
+          schools,
+          schools_users,
+          classroom_units
+        ]
+      }
+
+      it 'should include the top 10 results if they are present' do
+        runner = QuillBigQuery::TestRunner.new([
```

Comment:
This can be DRY'd up by leaning on the shared_context more, see my other comment.